### ISSUE TITLE
Can't get it to work

### ISSUE THREAD

**Issue by JacobDB** [dfrankland/hyper-sync-settings#30]
Sorry if this isn't the right place for this, but I can't seem to get this to work. 

I've create a personal access token with "gist" permission, and created a new secret gist. I copied the token and the gist ID in to the JSON file. I reloaded Hyper, and clicked on Plugins > Sync Settings > Backup. Nothing happens, no error, no notification, nothing.

I'm on Hyper 1.3.1 and hyper-sync-settings 2.0.2. Tried on two separate machines, one Mac, one PC, same thing. I'll try generating a new PAT and see if that does it.

**Comment by JacobDB** [comment]
Okay, so generating a new token actually did the trick... Not sure why, but whatever. I'm still having issues on my Windows machine, though. I can't restore the settings I uploaded from my Mac; again I get no error or anything. Double checked the token and gist ID are correct this time.

**Comment by dfrankland** [comment]
You updated the token on both machines? _Just double-checking 👍_

**Comment by JacobDB** [comment]
Yea, verified that my sync settings are identical on both machines.

**Comment by JacobDB** [comment]
I do see in ~/.hyper_plugins/.hyper-sync-settings that a .hyper.js file is getting copied, it's just not reading/writing from my GitHub... Will try generating a unique token for this machine in a bit, to see if that does it.

**Comment by JacobDB** [comment]
Yea, creating a new token specific for that device made no difference.

**Comment by dfrankland** [comment]
Do you have `git` installed on your Windows machine? If you do, can you go to `~/.hyper_plugins/.hyper-sync-settings` and perform a `git pull` to see what happens?

**Comment by JacobDB** [comment]
`fatal: Not a git repository (or any of the parent directories): .git`

**Comment by outaTiME** [comment]
me neither, im running on macos 10.12.4, hyper 1.3.1 and im unable to backup my config.
i create the personal access token, new gist and put it in .hyper-sync-settings.json

**Comment by realThinhIT** [comment]
Me neither, running Hyper on Windows 10 and can't get it to work.

Update 1: It couldn't update/ fetch data from GitHub.

**Comment by hardlyneutral** [comment]
Same issue here. OS X Sierra.

**Comment by dougmorin** [comment]
Also jumping on this issue.  Just installed everything as they suggested and it won't work.  Won't provide an error message, either.

**Comment by BartNetJS** [comment]
I disabled all my plugins, enabled only hyper-sync-settings and it starts working

**Comment by JacobDB** [comment]
@bartnetjs ah, so it's probably a conflict then. I'll see if that's the case for me as well.

**Comment by malonehedges** [comment]
Commenting to get notifications on this. Having the same vague issue

**Comment by PolGuixe** [comment]
Same issue here.

**Comment by itsfreddyrb** [comment]
Heya guys, I don't have a hyper-sync-settings.json file so I don't know the structure. Do I have to create one? Or something? Thanks.

**Comment by BartNetJS** [comment]
after installing the plugin and run hyper, type
cd %USERPROFILE%\.hyper_plugins\
you should find it there.
it is also the same structure as .hyper.js

**Comment by itsfreddyrb** [comment]
Yeah, it isn't there. *shrug* I've got this inside .hyper_plugins > local           node_modules    package.json

That's it!

Any help would be appreciated.

**Comment by dfrankland** [comment]
@freddyr0, if that file isn't generated that may be the issue other's are having as well. I'll look into it soon why that would be happening.

If that file doesn't exist, you can just copy the default file (<https://github.com/dfrankland/hyper-sync-settings/blob/master/src/config.default.json>) to `~/.hyper_plugins/.hyper-sync-settings.json`.

**Comment by itsfreddyrb** [comment]
@dfrankland yay! I helped! lol Yeah, that might be it, it simply isn't there, I tried on the imac running 10.12.4 and a macbook pro running the same.

**Comment by mikedevita** [comment]
alright i'm jumping in on this as well.. when i try to force a "Sync Settings -> Backup Settings" nothing happens. No errors or anything, and my gist doesn't get updated, gist is [here](https://gist.github.com/mikedevita/b28757d1fca0580b207618a60cd40dec) for example.

<img width="534" alt="screen shot 2017-05-31 at 8 39 41 am" src="https://cloud.githubusercontent.com/assets/2991640/26640444/950ac6a2-45dc-11e7-88d0-99eeb147023d.png">

<img width="499" alt="screen shot 2017-05-31 at 8 41 09 am" src="https://cloud.githubusercontent.com/assets/2991640/26640539/d8cbaf46-45dc-11e7-96ff-72eef42e736d.png">

**Comment by itsfreddyrb** [comment]
Yup. Same here. Btw, offtopic, any of you guys know of a good way to integrate the use of a mouse  with neo vim?

**Comment by ghost** [comment]
Same issue here

**Comment by scf4** [comment]
I get `Settings not found. Click for more info`

`cat ~/.hyper_plugins/.hyper-sync-settings.json` returns:

```
{
  "personalAccessToken": "[my token]",
  "gistId": "[my gist id]"
}
```

**Comment by jmfederico** [comment]
I was facing the same issue, and this is how I fixed it.

1. Make sure the only file you have on your gist is one named `.hyper.js`.
2. Manually copy the content of your local `.hyper.js`to the gist.
3. Modify you local file (any change will suffice)
4. Restore your settings from the gist `Plugins > Sync Settings > Restore Settings`

You should get a notification that your settings were restored. It should work now.

**Comment by YoEight** [comment]
Did what @jmfederico didn't work neither. Thanks for the help though.

**Comment by jpmckearin** [comment]
I am also experiencing similar issues and have attempted the "fixes" here. I took some time this morning to attempt to debug. It seems that `backup.js` is stopping [here](https://github.com/dfrankland/hyper-sync-settings/blob/68b8ab7484bb7896873becb24d1b2c8f7ec0a4d3/src/lib/getCommands/backup.js#L27) but is not throwing an error. I changed the `.hyper.js` settings and clicked `Plugins > Sync Settings > Backup Settings` which seemed to not do anything (no notifications of error/success). I then ran `git status` in `~/.hyper_plugins/.hyper-sync-settings` and saw that `.hyper.js` was modified and staged, but, not committed. I ran `git diff --cached` and saw the changes I made. I then ran `git fetch`, `git commit -m "[Date]"` and `git push origin master` without any issues. It seems this may be an issue with our usage of `simple-git`.

```
Hyper 1.4.2 (stable)
Plugins:
  hyper-sync-settings (2.0.2),
  hyperline (0.6.1),
```

**Comment by dfrankland** [comment]
@jpmckearin, nice work! I'll start looking into that. `simple-git` has been upgraded a few times since last release, so I would help that it will help fix this issue. If not, I think we should either just spawn git commands directly rather than use this library... Or maybe use Github's shiny new tool: [`dugite`](https://github.com/desktop/dugite)

**Comment by linonetwo** [comment]
Any update?

**Comment by raupie** [comment]
Reporting same issue with Windows 10 WSL Environment. Lack of error is frustrating.

**Comment by AddAndy** [comment]
I'm seeing the same issue.

Are there any error logs or debug logs i can read to see where the code is stopped?

**Comment by dustintheweb** [comment]
Also seeing this issue on Windows (WSL) - Backup Settings etc is not doing anything...

I was stuck for a while when `Plugins > Sync Settings` wasn't showing up in the menu, but a windows restart got it to show.

Now there is a `C:\Users\dusti\.hyper_plugins` folder with `.hyper.js` in it, but it is not changing when prompted by the menu option.

Anyone able to help?

**Comment by Arzaroth** [comment]
In my case the fix was quite easy (WSL here, although I doubt it makes any difference).
First, close hyper and create but a single file named `.hyper.js` in your gist with your settings.
Then, remove the `~/.hyper_plugins/.hyper-sync-settings` folder.
Lastly, launch hyper and `Plugins > Sync Settings > Restore Settings`.
It should clone your gist and you should have a notification saying everything went well.

**Comment by typerror** [comment]
In my case the problem was super easy to solve. 
I **forgot** to insert the plugin info inside my .hyper.js plugins array.

```
  plugins: ["hyperpower", "hyper-search", "hyper-sync-settings"],
```

BTW I'm giving my step by step installation since it's not very clear.

#### FIRST CREATE THE [GIST](https://gist.github.com)
```
- name it .hyper.js

- open your local ~/.hyper.js file either in the terminal or in a text editor

- copy and paste the code inside this file into the created gist

- keep your gist page open since you'll need it in a moment
```
#### SECOND CREATE A [TOKEN](https://github.com/settings/tokens) 
```
- thick the gist checkbox (gist permissions) 

- name it as you want. keep it memorable, simple and descriptive.

- copy the generated token in a temp file or don't refresh the page since you'll need the token in a sec
```
#### THEN CREATE THE SYNC-SETTING FILE
```
In the terminal

- cd ~/.hyper_plugins

- nano .hyper-sync-settings.json

- copy the following script inside .hyper-sync-settings.json
```
{
  "personalAccessToken": "my token", 
  "gistId": "my gist"
}

```
- replace "my token" with your personal access token from the GitHub webpage

- replace "my gist" with your personal gist id url from the gist webpage. 
   explanation: the gist id is the last part of the url after the username/from here over --> there

- your file should now look similar to the following
```
{
  "personalAccessToken": "81d55fs52be9fdb39746a785f9d1c7691947",
  "gistId": "81d55be9fdb39746a785f9d1c7691947"
}
```
- ctrl x

- y (save changes)
```

#### FINALLY INSTALL PLUGIN
```
- ~ (go home directory)

- cd ~/.hyper_plugins

- npm i hyper-sync-settings
```

#### AND LASTLY UPDATE YOUR HYPER.JS SETTINGS
```
- nano ~/.hyper.js

- type: 'ctrl w' and search 'plugins:'

- add "hyper-sync-settings" inside the array.

- example —> plugins: ["hyperpower", "hyper-search", "hyper-sync-settings"],

- crtl x

- y (save changes)
```

#### FULL RELOAD, CLOSE AND RELAUNCH HYPER

After that go to —> Plugins > Sync Settings > Backup Settings

![9cna0ceu5iuou](https://user-images.githubusercontent.com/32941768/46892894-4b034f80-ce6f-11e8-9c3e-31c0827d8923.gif)

You should receive a notification saying that your stuff were backed up!

You'll also note that you now have a new folder inside ~/.hyper-plugins/
```
- ls -la .hyper_plugins/.hyper-sync-settings
```

This is where there is our gist file namely '.hyper.js'

**Comment by bmitchinson** [comment]
🤔 Still struggling with this sorry.
My config is fine, and I used Arzaroth's method. It updates a copy of my .hyper.js locally, but it doesn't push to my gist. No errors are presented when I attempt to back up. Assuming there should be one to notify me that it didn't work?
Thanks for any advice.

**Comment by peter-avila** [comment]
I was able to get it working using @Arzaroth's suggestion then changing into the  `.hyper-sync-settings` directory then manually committing and pushing. Now I can backup, but everytime I make a change to my config via the `.hyper.js` file in root of my user profile directory, I get the attached error.

![image](https://user-images.githubusercontent.com/10645510/54101548-b6a80500-439a-11e9-8942-10e2ca5e307f.png)

**Comment by dfrankland** [comment]
Could you try the new `hyper-sync-settings@3.0.1`?

**Comment by ceane** [comment]
@Arzaroth I was able to get it to clone but not save 😔
